rasa/nlu/utils/__init__.py

```
@@ -315,6 +315,41 @@ def read_endpoints(endpoint_file: Text) -> "AvailableEndpoints":
     return AvailableEndpoints(model, data)
 
 
+def validate_pipeline_yaml(yaml):
+    """Validate NLU pipeline yaml."""
+    from pykwalify.core import Core
+    import pkg_resources
+    from pykwalify.errors import SchemaError
+    from ruamel.yaml import YAMLError
+    from rasa.nlu.model import InvalidModelError
+
+    log = logging.getLogger("pykwalify")
+    log.setLevel(logging.WARN)
+
+    try:
+        schema_file = pkg_resources.resource_filename(__name__, "schemas/nlu_model.yml")
+        source_data = rasa.utils.io.read_yaml(yaml)
+    except YAMLError:
+        raise InvalidModelError(
+            "The provided pipeline file is invalid. You can use "
+            "http://www.yamllint.com/ to validate the yaml syntax "
+            "of your domain file."
+        )
+
+    try:
+        c = Core(source_data=source_data, schema_files=[schema_file])
+        c.validate(raise_exception=True)
+    except SchemaError:
+        raise InvalidModelError(
```

Please raise `InvalidConfigError`